main/compiling/compilers/Compiler.py

```diff
@@ -579,20 +834,56 @@ def compile_final_detectors_from_measurements(
                     for pauli in detector_product_measured.paulis}
                 detector_qubits = sorted(
                     detector_qubits, key=lambda qubit: qubit.coords)
+
                 # Now calculate the product of the measurements on these qubits.
                 measurements = [
                     list(final_checks[data_qubit].paulis.values())[0]
                     for data_qubit in detector_qubits]
                 measurement_product = PauliProduct(measurements)
 
                 if detector_product_measured.equal_up_to_sign(measurement_product):
-                    # Can make a lid for this detector!
-                    floor = [
-                        (t + detector.lid_end, check)
-                        for t, check in detector_checks_measured]
-                    lid = [(0, final_checks[qubit]) for qubit in detector_qubits]
-                    final_detectors.append(Drum(floor, lid, 0, detector.anchor))
 
+                    floor = []
+                    for t, check in detector_checks_measured:
+
+                        expected_lid_end = (
+                            round//code.schedule_length)*code.schedule_length + detector.lid_end
+                        actual_lid_end = round
+
+                        if expected_lid_end < actual_lid_end:
+                            expected_lid_end += code.schedule_length
+                        difference = expected_lid_end - actual_lid_end
+
+                        floor_t = t + difference
+
+                        if floor_t == -1 + n:
+                            floor_t -= n
+
+                        floor.append((floor_t, check))
+
+                    lid = [(0, final_checks[qubit])
+                           for qubit in detector_qubits]
+
+                    new_drum = Drum(floor, lid, (round %
+                                    code.schedule_length), detector.anchor)
+
+                    # because gauge css floquet codes tries to add hexagonal detectors of consecutive rounds
+                    if isinstance(code, FloquetColourCode):
+                        if (new_drum.end - new_drum.start) == 1:
+                            pass
+                        else:
+                            final_detectors.append(new_drum)
+                    else:
+                        final_detectors.append(new_drum)
+
+        if add_small_detectors == True:
```

Comment:
Don't need to compare boolean values to True or False: just write `if add_small_detectors`